include/maths/CDataFrameRegressionModel.h

```diff
@@ -32,7 +32,7 @@ class MATHS_EXPORT CDataFrameRegressionModel {
     virtual ~CDataFrameRegressionModel() = default;
 
     //! Train the model on the values in \p frame.
-    virtual void train(core::CDataFrame& frame, TProgressCallback recordProgress = noop) = 0;
+    virtual void train(TProgressCallback recordProgress = noop) = 0;
```

Comment:
Also, `predict` still takes a data frame. I feel like this class should enforce that a `CDataFrame` has been supplied by requiring it by *reference* in the constructor. Then it can provide a protected accessor to the `CDataFrame` object. This forces any implementation to ensure that it is available at the point train is run.

Reply:
Done.